iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -32,6 +33,11 @@ static llvm::cl::opt<int> clNativeVectorSizeInBytes(
     llvm::cl::desc("native vector size to use on the hardware"),
     llvm::cl::init(16));
 
+static llvm::cl::opt<int> clNumberOfRuntimeThreads(
```

Comment:
I'll add a TODO. Unfortunately this needs to be static "guess". If this is not static, none of the vectorization stuff will kick in. So if there is a way to provide "hint" for now, then that'd be useful. I'll add a TODO here for now.

100% agree this can become load bearing. I actually added it for your benefit since you were experimenting with things being more partitioned for the stream work. For now its assuming a default of 8 threads.

Reply:
Yep yep - for now anything goes - but the next step here will be separating what you need for vectorization vs what you need to compute workgroup count - they should be at different orders of magnitude and not practically interact (besides indirectly in some future cost model magic) - for example, whether you decide to vectorize with 4 elements in a loop N*4 times or 16 elements in a loop N times doesn't mean you need to know N at compile time: you'd still be processing the same number of elements per workgroup. There's of course derivative details there (register/cache pressure, etc) but those are independent from vectorization being possible and just shades of detail that can be left to PGO/search vs something structural and flexible like this.

For this particular change I'm curious as to why we need to conflate the two - but "because this is just a step in the right direction and not _the_ solution" is totally fine :)

Reply:
(oh and lgtm with a todo/scary warning, mostly just curious)

Reply:
There is some unfortunate coupling that happens because of everything required to be static as we vectorize. If things arent static at the first level of tiling, then they cant be static at the next etc. etc. In time we will decouple this with peeling, and maybe `assume` intrinsics which can aid getting to the static vector sizes when we need. A lot of this is just working around this. The next big step in codegen is to handle dynamic shape codegen effectively. Then all of this will be much more decoupled.